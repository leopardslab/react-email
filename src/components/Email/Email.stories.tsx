import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Email } from './Email';

export default {
  component: Email,
} as ComponentMeta<typeof Email>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Email> = (args) => <Email {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: (
    <table style={{ width: '100%' }}>
      <tbody>
        <tr>
          <td>
            <p style={{ margin: '0', fontSize: '30px' }}>Hello World</p>
          </td>
        </tr>
      </tbody>
    </table>
  ),
  classes: {
    root: {
      backgroundColor: 'gray',
      color: 'white',
    },
  },
};
