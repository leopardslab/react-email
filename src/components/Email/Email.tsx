import { ReactNode } from 'react';
import { BaseStyleProp } from '../types';
import { makeStyles } from '../../utils/makeStyles';

type EmailStyles = 'root';

export interface EmailProps extends BaseStyleProp<EmailStyles> {
  children?: ReactNode;
}

const useStyles = makeStyles({
  root: {
    margin: '0px auto',
    maxWidth: '600px',
  },
});

export const Email = ({ children, className, classes }: EmailProps): JSX.Element => {
  const styles = useStyles({ classes });

  return (
    <div style={styles.root} className={className}>
      {children}
    </div>
  );
};
