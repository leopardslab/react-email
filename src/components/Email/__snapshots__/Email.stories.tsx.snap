// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Email Default smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px; background-color: lightgray; color: white;">
  <table style="width: 100%;">
    <tbody>
      <tr>
        <td>
          <p style="margin: 0px; font-size: 30px;">
            Hello World
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
